src/pages/AddressInfoPage/index.tsx

```diff
@@ -0,0 +1,325 @@
+  const [txLoading, setTxLoading] = useState(true)
+
+  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
+
+  const [exportModalShown, setExportModalShown] = useState(false)
+
+  // Default page
+  const pageNumber = usePageNumber()
+
+  useEffect(() => {
+    if (!client || !id) return
+
+    const fetch = async () => {
+      setAddressDataLoading(true)
+      setAddressData(undefined)
+      try {
+        const addressData = await fetchAddressData(client, id)
+        setAddressData(addressData)
+      } catch (error) {
+        console.error(error)
+        setSnackbarMessage({
+          text: getHumanReadableError(error, 'Error while fetching address data'),
+          type: 'alert'
+        })
+      } finally {
+        setAddressDataLoading(false)
+      }
+    }
+
+    fetch()
+  }, [client, id, setSnackbarMessage])
+
+  useEffect(() => {
+    if (!client || !id) return
+
+    const fetch = async () => {
+      setTxLoading(true)
+      setAddressTransactions(undefined)
+      try {
+        const transactionsData = await fetchAddressTransactions(client, id, pageNumber)
+        setAddressTransactions(transactionsData)
+      } catch (error) {
+        console.error(error)
+        setSnackbarMessage({
+          text: getHumanReadableError(error, "Error while fetching address' transactions"),
+          type: 'alert'
+        })
+      } finally {
+        setTxLoading(false)
+      }
+    }
+
+    fetch()
+  }, [client, id, pageNumber, setSnackbarMessage])
+
+  // Asset price (appox).
+  // TODO: when listed tokens, add resp. prices. ALPH only for now.
+  useEffect(() => {
+    setAddressWorth(undefined)
+
+    const getAddressWorth = async () => {
+      const balance = addressData?.details.balance
+      if (!balance) return
+
+      const price = await fetchAssetPrice('alephium')
+
+      setAddressWorth(calculateAmountWorth(BigInt(balance), price))
+    }
+    getAddressWorth()
+  }, [addressData?.details.balance])
+
+  const handleExportModalOpen = () => setExportModalShown(true)
+  const handleExportModalClose = () => setExportModalShown(false)
+
+  if (!id) return null
+
+  const addressGroup = addressToGroup(id, TOTAL_NUMBER_OF_GROUPS)
+
+  const totalBalance = addressData?.details.balance
+  const lockedBalance = addressData?.details.lockedBalance
+  const txNumber = addressData?.details.txNumber
+  const txList = addressTransactions?.transactions
+  const latestActivityDate = txList?.[0]?.timestamp
+
+  let assets = addressData?.tokens.map((t) => ({
+    ...t,
+    balance: BigInt(t.balance),
+    lockedBalance: BigInt(t.lockedBalance),
+    ...TokensMetadata[networkType].tokens.find((tm) => tm.id === t.id)
+  })) as Asset[]
+
+  if (totalBalance && lockedBalance && parseInt(totalBalance) > 0) {
+    assets?.push({ ...ALPH, balance: BigInt(totalBalance), lockedBalance: BigInt(lockedBalance) })
+  }
+
+  if (assets) {
+    assets = sortBy(assets, 'name')
+  }
```

Comment:
Not so happy about this part (declaring conditional variables + bit of processing), but I didn't want to spend much more time on this and risking overengineering. If you have any proposition, I'm all ears 🤝

Reply:
What about this?

```tsx
const assets = (addressData?.tokens.map((t) => ({
    ...t,
    balance: BigInt(t.balance),
    lockedBalance: BigInt(t.lockedBalance),
    ...TokensMetadata[networkType].tokens.find((tm) => tm.id === t.id)
})) ?? []) as Asset[]

if (totalBalance && lockedBalance && parseInt(totalBalance) > 0) {
  assets.push({ ...ALPH, balance: BigInt(totalBalance), lockedBalance: BigInt(lockedBalance) })
}

....

<AssetList assets={sortBy(assets, 'name')} isLoading={addressDataLoading} />
```

Reply:
That's already something, thanks.